chef/cookbooks/network/recipes/default.rb

```diff
@@ -393,6 +393,20 @@ def get_datapath_id_for_ovsbridge(bridge)
     nic.tx_offloading = node["network"]["enable_tx_offloading"] || true
   end
 
+  # if the nic is a vlan and the physical parent is not used, also set
+  # the mtu from the vlan nic for the parent
+  if nic.kind_of?(Nic::Vlan)
+    unless ifs[nic.parent].key? "mtu"
+      parent_nic = Nic.new(nic.parent)
+      # if there are multiple vlan's on the parent nic, use the highest mtu
+      if parent_nic.mtu < ifs[nic.name]["mtu"].to_i
+        Chef::Log.info("vlan #{nic.name} wants mtu #{nic.mtu} but parent #{parent_nic.name} " \
+                       "has no explicit mtu. Set mtu for #{parent_nic.name} to #{ifs[nic.name]['mtu']}")
+        parent_nic.mtu = ifs[nic.name]["mtu"]
```

Comment:
In order to have the new mtu also appear in the config files you need to set ```ifs[nic.parent]["mtu"] = ifs[nic.name]["mtu"]``` as well.